src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2017 Gabriel Loewen
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.durabletask;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Plugin;
+import hudson.Launcher;
+import jenkins.model.Jenkins;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.File;
+import hudson.model.TaskListener;
+import java.io.IOException;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Runs a Powershell script
+ */
+public final class PowershellScript extends FileMonitoringTask {
+    private final String script;
+    private boolean capturingOutput;
+
+    @DataBoundConstructor public PowershellScript(String script) {
+        this.script = script;
+    }
+    
+    public String getScript() {
+        return script;
+    }
+
+    @Override public void captureOutput() {
+        capturingOutput = true;
+    }
+
+    @SuppressFBWarnings(value="VA_FORMAT_STRING_USES_NEWLINE", justification="%n from master might be \\n")
+    @Override protected FileMonitoringController doLaunch(FilePath ws, Launcher launcher, TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
+        List<String> args = new ArrayList<String>();
+        PowershellController c = new PowershellController(ws);
+        
+        String cmd;
+        if (capturingOutput) {
+            // Using redirection in PowerShell produces extra newlines in output, so I am using [io.file]::WriteAllText to prevent corrupted output of exit code
+            cmd = String.format("$PSDefaultParameterValues = @{\"Out-File:Encoding\"=\"ASCII\"}; & \"%s\" *> \"%s\" 2> \"%s\"; [io.file]::WriteAllText(\"%s\",$LastExitCode); [io.file]::AppendAllText(\"%s\",$error);", 
```

Comment:
For the output (non-error) you could just pipe to "Out-File -Encoding UTF8" instead of redirecting directly to a file, but the error output would still have the UTF-16 though.
Something like (untested)
```
String.format("& \"%s\" *>1 2> \"%s\" | Out-File -Encoding \"UTF-8\" \"%s\"; [io.file]::WriteAllText(\"%s\",$LastExitCode); [io.file]::AppendAllText(\"%s\",$error);", 
```